mlir/lib/Dialect/MIOpen/Transforms/AffixTuningParameters.cpp

```diff
@@ -201,13 +203,13 @@ void AffixTuningParameters::affixTuningParametersImpl(T &op) {
       signalPassFailure();
     }
 
-    if (launchDimCallback) {
-      launchDimCallback(validParams.blockSize, gridSize);
-    }
     op.setAttr("m_per_thread", b.getI32IntegerAttr(validParams.gemmMPerThread));
     op.setAttr("n_per_thread", b.getI32IntegerAttr(validParams.gemmNPerThread));
     op.setAttr("block_size", b.getI32IntegerAttr(validParams.blockSize));
 
+    func.setAttr("block_size", b.getI32IntegerAttr(validParams.blockSize));
+    func.setAttr("grid_size", b.getI32IntegerAttr(gridSizeOverride ? gridSizeOverride : gridSize));
```

Comment:
Up to now, `func` is still a cpu function. Conceptually speaking, having a cpu function to have attributes of blockSize and gridSize can be confusing, and that having ops to carry this information is more reliable on the long run. However, this cpu function does get converted directly to a gpu function and this change is simpler than either attributes to be part of `conv2d` op or `gridwisegemm` op.

What do you think @whchung?

Reply:
Yes, that was what I was seeing that the function became the gpu function (easily extracted in populateKernelLaunch). It is easy to pass along on the actual Op and maybe further down the pipeline attach to the explicit gpu function. Not sure where that would be..

Reply:
On a second thought, `gridwisegemm` is not a good candidate to carry grid information. On the other hand, the conversion from cpu to gpu function happens at `LowerMIOpenOpsToGPUPass` and is also reasonable. I think you changes look good.